pysrc/bytewax/inputs.py

```diff
@@ -0,0 +1,204 @@
+"""Helpers to let you quickly define epoch / batching semantics.
+
+Use these to wrap an existing iterator which yields items.
+"""
+import datetime
+import heapq
+from operator import itemgetter
+from typing import Any, Iterable, Tuple, Callable
+
+
+def single_batch(wrap_iter: Iterable) -> Iterable[Tuple[int, Any]]:
+    """All input items are part of the same epoch.
+
+    Use this for non-streaming-style batch processing.
+
+    >>> list(single_batch(["a", "b", "c"]))
+    [(0, 'a'), (0, 'b'), (0, 'c')]
+
+    Args:
+
+        wrap_iter: Existing input iterable of just items.
+
+    Yields: Tuples of (epoch, item).
+    """
+    for item in wrap_iter:
+        yield (0, item)
+        
+
+def order(
```

Comment:
This is great, I think this is a good way to handle late-arriving and out-of-order data on the input side of things. TIL about heapq built-in